spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubOperations.java

```diff
@@ -1,102 +1,34 @@
-	 * @return the listenable future of the call
-	 */
-	ListenableFuture<String> publish(String topic, PubsubMessage pubsubMessage);
-
-	/**
-	 * Add a callback method to an existing subscription.
-	 *
-	 * <p>The created {@link Subscriber} is returned so it can be stopped.
-	 * @param subscription the name of an existing subscription
-	 * @param messageHandler the callback method triggered when new messages arrive
-	 * @return subscriber listening to new messages
-	 */
-	Subscriber subscribe(String subscription, MessageReceiver messageHandler);
-
-	/**
-	 * Pull and auto-acknowledge a number of messages from a Google Cloud Pub/Sub subscription.
-	 * @param subscription the subscription name
-	 * @param maxMessages the maximum number of pulled messages
-	 * @param returnImmediately returns immediately even if subscription doesn't contain enough
-	 * messages to satisfy {@code maxMessages}
-	 * @return the list of received messages
-	 */
-	List<PubsubMessage> pullAndAck(String subscription, Integer maxMessages,
-			Boolean returnImmediately);
-
-	/**
-	 * Pull a number of messages from a Google Cloud Pub/Sub subscription.
-	 * @param subscription the subscription name
-	 * @param maxMessages the maximum number of pulled messages
-	 * @param returnImmediately returns immediately even if subscription doesn't contain enough
-	 * messages to satisfy {@code maxMessages}
-	 * @return the list of received acknowledgeable messages
-	 */
-	List<AcknowledgeablePubsubMessage> pull(String subscription, Integer maxMessages,
-			Boolean returnImmediately);
-
-	/**
-	 * Pull and auto-acknowledge a message from a Google Cloud Pub/Sub subscription.
-	 * @param subscription the subscription name
-	 * @return a received message, or {@code null} if none exists in the subscription
-	 */
-	PubsubMessage pullNext(String subscription);
-}
+/*
+ *  Copyright 2017-2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.core;
+
+import org.springframework.cloud.gcp.pubsub.core.publisher.PubSubPublisherOperations;
+import org.springframework.cloud.gcp.pubsub.core.subscriber.PubSubSubscriberOperations;
+
+/**
+ * An abstraction for Google Cloud Pub/Sub for publish and subscribe / pull methods
+ *
+ * @author Vinicius Carvalho
+ * @author João André Martins
+ * @author Mike Eltsufin
+ * @author Chengyuan Zhao
+ * @author Doug Hoard
+ * @since 1.0.0

```

Comment:
We don't need `@since 1.0.0` because it is really obvious that the first version of the API is from the `1.0`.
More over we don't use the third number in the version if it is `0`.

On the other hand: something happened to indents in this class?
Do you still stick with tabs, not 4 spaces?

Reply:
Removed `@since 1.0.0` since it's obvious.  I am not seeing an issue with the tabs vs spaces.  I have setup my idea to use tabs.  The checkstyle configuration appears to catch indents with spaces ... so I'm not sure what you are seeing.